### ISSUE TITLE
SAGA version 7.3.0 is not officially supported - algorithms may encounter issues

### ISSUE THREAD

**Issue by mapninja** [qgis/QGIS-Mac-Packager#94]
Some SAGA tools not working, including Raster Calculator, Simple Filter variants, etc...

Mac Catalina: 10.15.7

QGIS version | 3.15.0-Master | QGIS code revision | 252976b739
-- | -- | -- | --
Compiled against Qt | 5.14.2 | Running against Qt | 5.14.2
Compiled against GDAL/OGR | 3.1.2 | Running against GDAL/OGR | 3.1.2
Compiled against GEOS | 3.8.1-CAPI-1.13.3 | Running against GEOS | 3.8.1-CAPI-1.13.3
Compiled against SQLite | 3.31.1 | Running against SQLite | 3.31.1
PostgreSQL Client Version | 12.3 | SpatiaLite Version | 4.3.0a
QWT Version | 6.1.4 | QScintilla2 Version | 2.11.4
Compiled against PROJ | 6.3.2 | Running against PROJ | Rel. 6.3.2, May 1st, 2020
OS Version | macOS 10.15
Active python plugins | processing; db_manager; MetaSearch

Raster Calculator:

2020-10-20T16:31:08     INFO    SAGA execution commands
             io_gdal 0 -TRANSFORM 1 -RESAMPLING 3 -GRIDS "/private/var/folders/f3/0prmkq7962795yfpwsmvh2100000gq/T/processing_TduLpX/48cd038e0bb74eeca557843bce17eb19/MardemSlope.sgrd" -FILES "/Users/maples/Downloads/L10/MardemSlope.tif"
             grid_calculus "Grid Calculator" -GRIDS "/private/var/folders/f3/0prmkq7962795yfpwsmvh2100000gq/T/processing_TduLpX/48cd038e0bb74eeca557843bce17eb19/MardemSlope.sgrd" -FORMULA "int(a)" -RESAMPLING 3 -USE_NODATA false -TYPE 7 -RESULT "/Users/maples/Downloads/L10/SlopeInt.sdat"
2020-10-20T16:31:08     INFO    SAGA execution console output
             dyld: Symbol not found: __cg_jpeg_resync_to_restart
             Referenced from: /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
             Expected in: /Applications/QGISNightly.app/Contents/MacOS/lib/libJPEG.dylib
             in /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
             /Users/maples/Library/Application Support/QGIS/QGIS3/profiles/NewNightly/processing/saga_batch_job.sh: line 3: 23840 Abort trap: 6 saga_cmd io_gdal 0 -TRANSFORM 1 -RESAMPLING 3 -GRIDS "/private/var/folders/f3/0prmkq7962795yfpwsmvh2100000gq/T/processing_TduLpX/48cd038e0bb74eeca557843bce17eb19/MardemSlope.sgrd" -FILES "/Users/maples/Downloads/L10/MardemSlope.tif"
             dyld: Symbol not found: __cg_jpeg_resync_to_restart
             Referenced from: /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
             Expected in: /Applications/QGISNightly.app/Contents/MacOS/lib/libJPEG.dylib
             in /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
             /Users/maples/Library/Application Support/QGIS/QGIS3/profiles/NewNightly/processing/saga_batch_job.sh: line 4: 23841 Abort trap: 6 saga_cmd grid_calculus "Grid Calculator" -GRIDS "/private/var/folders/f3/0prmkq7962795yfpwsmvh2100000gq/T/processing_TduLpX/48cd038e0bb74eeca557843bce17eb19/MardemSlope.sgrd" -FORMULA "int(a)" -RESAMPLING 3 -USE_NODATA false -TYPE 7 -RESULT "/Users/maples/Downloads/L10/SlopeInt.sdat"

QGIS version: 3.15.0-Master
QGIS code revision: 252976b739
Qt version: 5.14.2
GDAL version: 3.1.2
GEOS version: 3.8.1-CAPI-1.13.3
PROJ version: Rel. 6.3.2, May 1st, 2020
Processing algorithm…
Algorithm 'Simple filter' starting…
Input parameters:
{ 'INPUT' : '/Users/maples/Downloads/L10/MardemSlope.tif', 'METHOD' : 0, 'MODE' : 0, 'RADIUS' : 3, 'RESULT' : '/Users/maples/Downloads/L10/SlopeFilter.sdat' }

io_gdal 0 -TRANSFORM 1 -RESAMPLING 3 -GRIDS "/private/var/folders/f3/0prmkq7962795yfpwsmvh2100000gq/T/processing_TduLpX/6f290333263c4262b1bc09f4bd30e665/MardemSlope.sgrd" -FILES "/Users/maples/Downloads/L10/MardemSlope.tif"
grid_filter "Simple Filter" -INPUT "/private/var/folders/f3/0prmkq7962795yfpwsmvh2100000gq/T/processing_TduLpX/6f290333263c4262b1bc09f4bd30e665/MardemSlope.sgrd" -MODE 0 -METHOD 0 -RADIUS 3 -RESULT "/Users/maples/Downloads/L10/SlopeFilter.sdat"
dyld: Symbol not found: __cg_jpeg_resync_to_restart
Referenced from: /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
Expected in: /Applications/QGISNightly.app/Contents/MacOS/lib/libJPEG.dylib
in /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
/Users/maples/Library/Application Support/QGIS/QGIS3/profiles/NewNightly/processing/saga_batch_job.sh: line 3: 23872 Abort trap: 6 saga_cmd io_gdal 0 -TRANSFORM 1 -RESAMPLING 3 -GRIDS "/private/var/folders/f3/0prmkq7962795yfpwsmvh2100000gq/T/processing_TduLpX/6f290333263c4262b1bc09f4bd30e665/MardemSlope.sgrd" -FILES "/Users/maples/Downloads/L10/MardemSlope.tif"
dyld: Symbol not found: __cg_jpeg_resync_to_restart
Referenced from: /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
Expected in: /Applications/QGISNightly.app/Contents/MacOS/lib/libJPEG.dylib
in /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
/Users/maples/Library/Application Support/QGIS/QGIS3/profiles/NewNightly/processing/saga_batch_job.sh: line 4: 23873 Abort trap: 6 saga_cmd grid_filter "Simple Filter" -INPUT "/private/var/folders/f3/0prmkq7962795yfpwsmvh2100000gq/T/processing_TduLpX/6f290333263c4262b1bc09f4bd30e665/MardemSlope.sgrd" -MODE 0 -METHOD 0 -RADIUS 3 -RESULT "/Users/maples/Downloads/L10/SlopeFilter.sdat"
Execution completed in 0.40 seconds
Results:
{'RESULT': '/Users/maples/Downloads/L10/SlopeFilter.sdat'}

Loading resulting layers
The following layers were not correctly generated.
• /Users/maples/Downloads/L10/SlopeFilter.sdat
You can check the 'Log Messages Panel' in QGIS main window to find more information about the execution of the algorithm.

**Comment by gioman** [comment]
Isn't this the same as https://github.com/qgis/QGIS-Mac-Packager/issues/93 ?

**Comment by PeterPetrik** [comment]
@gioman no, this one is library  mismatch between system and bundled libJPEG. the other one is that SAGA changed API for some algorithms

**Comment by gotjoshua** [comment]
not sure what is needed in the qgis internals to make this work (otherwise i'd submit a PR)

but i found a workaround strategy and posted it here:  
https://gis.stackexchange.com/a/398282/81215
